156164454 check for multi doc types

davidbl opened the pull request:
Pivotal Tracker Story: https://www.pivotaltracker.com/story/show/156164454

divoxx commented on lib/elasticity/index_config.rb:
```diff
@@ -1,5 +1,9 @@
   class IndexConfig
+    class SubclassError < StandardError; end
+
+    SUBCLASSES_NOT_AVAILABLE = "subclasses are not available in this version of Elasticsearch".freeze
```
Minor thing but I think it's important to give a more informative error message. What you think of: `Document type support have been removed from Elasticsearch 6, therefore document type based inheritance has been disabled by Elasticity`

divoxx commented on lib/elasticity/index_config.rb:
```diff
@@ -14,6 +18,7 @@ def initialize(elasticity_config, defaults = {})
       end
       @elasticity_config = elasticity_config
       yield(self)
+      subclasses_warning
```
I think this is a little more complicated than it needs to. We don't need a warning and an error, and ideally the `IndexMapper` class shouldn't need to call methods on the config when creating/recreating the index. If we simply raise the error here, it will prevent the class from being configured and force the user to make the correction rather than catching it in runtime.

davidbl commented:
bruno  asked for  the warnings and errrors

@brupm   want to chime in on Rodger's question?

brupm commented:
@divoxx the reason for the warning is that technically until es7 searching multi doctype indexes is supported, however writing to them is no longer possible in 6. So I thought it would be nice to properly differentiate it - don't feel strongly about it.

divoxx commented:
@brupm @davidbl Ok, I didn't know that and it makes sense. But I think the implementation should make that more clear, both in the code and to the user.

To keep things simple, I think both should occur on the same place, during the initialization of the `IndexConfig` and then just conditionally showing the warning for ES6 and an error for ES7, with different messages explaining the reason.

davidbl commented:
@brupm @divoxx 

I have simplified the logic and changed the message(s) to more closely match  the text here https://www.elastic.co/guide/en/elasticsearch/reference/6.x/removal-of-types.html#removal-of-types
